lib/data_hygiene/publishing_api_document_republisher.rb

```diff
@@ -0,0 +1,36 @@
+module DataHygiene
+  # Can be used to republish types of Editions to the publishing-api.
+  #
+  # Usage:
+  #
+  #   publisher = DataHygiene::PublishingApiDocumentRepublisher.new(CaseStudy)
+  #   publisher.perform
+  class PublishingApiDocumentRepublisher
+    attr_reader :edition_class, :logger, :queued
+
+    def initialize(edition_class, logger=Logger.new(STDOUT))
+      unless edition_class.is_a?(Class) && edition_class < Edition # http://ruby-doc.org/core-2.3.0/Module.html#method-i-3C
+        raise ArgumentError, "The argument to PublishingApiDocumentRepublisher must be a subclass of Edition"
+      end
+      @edition_class = edition_class
+      @logger = logger
+      @queued = 0
+    end
+
+    def perform
+      logger.info "Queuing #{documents.count} #{edition_class} instances for republishing to the Publishing API"
+      documents.each do |document|
```

Comment:
`find_each` may be preferable for batching. For smaller sets it has no discernible overhead.